agent/checks/os_service_windows.go

```diff
@@ -0,0 +1,114 @@
+//go:build windows
+// +build windows
+
+package checks
+
+import (
+	"fmt"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+var (
+	win windowsSystem = windowsOS{}
+)
+
+type OSServiceClient struct {
+	scHandle windows.Handle
+}
+
+func NewOSServiceClient() (*OSServiceClient, error) {
+	var s *uint16
+	scHandle, err := win.OpenSCManager(s, nil, windows.SC_MANAGER_CONNECT)
+
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to service manager: %w", err)
+	}
+
+	return &OSServiceClient{
+		scHandle: scHandle,
+	}, nil
+}
+
+func (client *OSServiceClient) Check(serviceName string) (err error) {
+	var isHealthy bool
+
+	m := win.getWindowsSvcMgr(client.scHandle)
+	defer func() {
+		errDisconnect := m.Disconnect()
+		if isHealthy || errDisconnect == nil || err != nil {
+			return
+		}
+		//unreachable at the moment but we might want to log this error. leaving here for code-review
+		err = errDisconnect
+	}()
+
+	svcHandle, err := win.OpenService(win.getWindowsSvcMgrHandle(m), syscall.StringToUTF16Ptr(serviceName), windows.SC_MANAGER_ENUMERATE_SERVICE)
+	if err != nil {
+		return fmt.Errorf("error accessing service: %w", err)
+	}
+	service := win.getWindowsSvc(serviceName, svcHandle)
+	defer func() {
+		errClose := service.Close()
+		if isHealthy || errClose == nil || err != nil {
+			return
+		}
+		//unreachable at the moment but we might want to log this error. leaving here for code-review
+		err = errClose
+	}()
+	status, err := service.Query()
+	if err != nil {
+		return fmt.Errorf("error querying service status: %w", err)
+	}
```

Comment:
This is a bit confusing for me. I don't know how in details the windows library work, but looking to this, `getWindowsSvc` will only allocate the struct and don't do any access, right? in that case what will happen if Query fail for some reason but we still call `service.Close` in the defer? will `service.Close` error?